docs/stack/client/querying.md

```diff
@@ -0,0 +1,164 @@
+
+    // Get all proposals' Id of the DAO
+    dao.proposals().subscribe(() => {})
+
+#### Raw GraphQL queries
+These queries are made using the static method `arc.sendQuery`. Pass the query designed [above](#raw-graphql-queries) as the parameter to `sendQuery`.
+
+    // Get votes id and outcome of the given Proposal
+    arc.sendQuery(gql`query {
+        proposal (id: "0x1245") {
+          votes { id outcome }
+        }}`)
+
+
+### Subscribing Query
+*Subscriptions* are used so that our handler gets called every time a new value is emitted for the observable stream. This is useful to keep the app data updated as the value changes.
+
+#### Entity methods
+As we saw the Entity methods do not send the query to the server but return an observable. We must subscribe as follows to send the query as well as subscribe for updates.
+
+```
+arc.daos({subscribe: true}).subscribe(() => {})
+dao.state({subscribe: true}).subscribe( () => {})
+```
+  
+#### Raw graphQL queries
+
+For even more control over what data is being fetched and subscribed to, you can write explicit queries:
+
+```
+arc.subscribe(gql`subscribe { proposal (id: "0x1245") { votes { id outcome }})`)
+new Proposal("0x1245").votes().subscribe(
+  (next) => {
+
+  }
+)
+```
+
+## Optimizing Subscription and Using Cache
+Since, subscriptions can be expensive, this behavior can be controlled/optimized in several ways.
+  
+  - [Controlling fetchPolicy](#controlling-apollo-fetchpolicy): controlling cache interaction.
+  - [Subscribing to Apollo Cache](#subscribe-to-apollo-cache-changes): getting updates from Apollo cache instead of graphnode server.
+  - [FetchAllData and Nested subscription](#use-fetchalldata-with-nested-subscription): by querying larger set at top level and subscribing to Apollo cache for nested queries.
+
+### Controlling Apollo fetchPolicy
+We can pass Apollo's [`fetchPolicy`](https://www.apollographql.com/docs/react/api/react-apollo/#optionsfetchpolicy) argument to control how the query interacts with the cache:
+
+  - _**cache-first**_: default value. Read data from cache first, fetch from network otherwise.
+  - _**cache-and-network**_: return data from cache but still always fetch from network. It optimizes quick response while also keep cached data updated.
+  - _**network-only**_: will always make a request network and write data to cache. It optimizes for data consistency with the server.
+  - _**cache-only**_: will never execute a query using your network interface and throw error if data not available in cache.
+  - _**no-cache**_: always make a request using your network interface. Also, it will not write any data to the cache
+
+```
+arc.daos({}, { fetch-policy: 'cache-first'}) // the default value
+arc.daos({where: {stage: "Boosted"}}, { fetch-policy: 'network-only'}) // bypass the cache
+```
+
+### Subscribe to Apollo Cache changes
```

Comment:
This section seems redundant.  All this should already be explained elsewhere.  In any case, I wouldn't introduce this name "Apollo Cache".  There is only one cache.

Reply:
> "Subscribe to the cache"
> 
> What cache? Client lib? Apollo?

Earlier you had commented which cache ... I'm little confuse

Reply:
It is little redundant but I think should be referred at-least briefly in optimization section.

Reply:
If you want to have this section, then rather than duplicate the information, I would replace the redundancy with a link to the place where it is explained.

Regarding the "Apollo Cache", I don't think we should refer to this unless we define it somewhere, and distinguish clearly between it and the client lib cache.  I don't honestly know the answer to that.  @jellegerbrandy

Reply:
Added link to Apollo (as pointed by @jellegerbrandy) and brief description.